### PR TITLE
feat: add helper function to keep track of the tutorial usage

### LINKED ISSUE #4189: [FEATURE] create a telemetry integration for keeping track of tutorials
it might be nice to add an [optional telemetry module](https://haystack.deepset.ai/tutorials/14_query_classifier) to see the number of usages for tutorials.

### PR DESCRIPTION
**Pull request by plaguss**
# Description

This PR includes a function to keep track of the tutorials running, similar to the format used in haystack:

```python
from argilla.utils.telemetry import tutorial_running

tutorial_running("tutorial_filename")
```

Closes #4189

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [x] I followed the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by plaguss** [comment]
@frascuchon, @davidberenstein1957, what do you think?
The idea is like the one used in [haystack's](https://haystack.deepset.ai/tutorials/14_query_classifier) as suggested by David. Would we need more information?

I see we already have [telemetry docs](https://docs.argilla.io/en/latest/reference/telemetry.html) so we could just reference that in the tutorials.

**Comment by davidberenstein1957** [comment]
I think we might also just use `Path(__file__).name`?

**Comment by plaguss** [comment]
> I think we might also just use `Path(__file__).name`?

To keep track of the tutorial instead of the *int* id?

**Comment by davidberenstein1957** [comment]
@plaguss, could you align with one of the interns about this usage in the tutorial and potentially we want to do some Argilla version check?

**Comment by davidberenstein1957** [comment]
@plaguss, perhaps using `__file__` out of the box woul be better to not allow user to pass any argument?

**Comment by plaguss** [comment]
> @plaguss, perhaps using `__file__` out of the box woul be better to not allow user to pass any argument?

I saw that `__file__` wasn't working on the notebook, and it seems that the variable is not available:
- https://stackoverflow.com/questions/39125532/file-does-not-exist-in-jupyter-notebook,
- https://github.com/ipython/ipython/issues/10123
Others tried to work with the `cwd`, but that would not work either... Do you have a better solution to avoid this?

**Comment by plaguss** [comment]
@davidberenstein1957 could you check the last addition? I tested the function in the different environments and seems to work... but needs an extra dependency: [`ipynbname`](https://pypi.org/project/ipynbname/). Let's me know what you think (if we want this option we should add the dependency).

File: src/argilla/utils/telemetry.py [hunk 1]
```diff
@@ -103,13 +103,43 @@ async def track_login(request: "Request", username: str):
...
+        import urllib.parse
+
+        import ipynbname
+
+        return Path(urllib.parse.unquote_plus(ipynbname.name())).stem
+    except Exception as e:
+        import warnings
+
+        warnings.warn("Couldn't determine the filename.")
+        return
```
Comment (davidberenstein1957) [inline src/argilla/utils/telemetry.py]: I think we can remove this `Return`

**Review by davidberenstein1957 — COMMENTED** [review]
Hi, thanks @plaguss . Works like a charm :)

File: src/argilla/utils/telemetry.py (hunk 1 above)
Comment (plaguss) [reply src/argilla/utils/telemetry.py]: Done!

File: src/argilla/utils/telemetry.py [hunk 2]
```diff
@@ -103,5 +103,39 @@ async def track_login(request: "Request", username: str):
...
+            globals_ = globals()
+            return Path(globals_["__vsc_ipynb_file__"]).stem
+    except KeyError as e:
+        # This should work for notebooks running locally or using google colab
+        import urllib.parse
+
+        import ipynbname
+
+        return Path(urllib.parse.unquote_plus(ipynbname.name())).stem
+
```
Comment (davidberenstein1957) [inline src/argilla/utils/telemetry.py:132]: I think we should have a final except to not break the user when it does not run.

Comment (plaguss) [reply src/argilla/utils/telemetry.py:132]: With the last change it won't break, instead it wont track the tutorial (without warning at all). Do we want this behavior?

Comment (davidberenstein1957) [reply src/argilla/utils/telemetry.py:132]: Yes, than it is alright.

**Pull request merged**